Applications/App/qAppWelcomeScreen.cxx

```diff
@@ -250,6 +250,17 @@ void qAppWelcomeScreen::loadModule(const QString& moduleName, int layout)
   slicerMainWindow->moduleSelector()->selectModule(moduleName);
   emit done();
 }
+//-----------------------------------------------------------------------------
+void qAppWelcomeScreen::loadNewData()
+{
+	Q_D(qAppWelcomeScreen);
+	qSlicerIOManager *ioManager = qSlicerApplication::application()->ioManager();
+	if (!ioManager)
```

Comment:
Style:

Please follow the Slicer style here: http://www.slicer.org/slicerWiki/index.php/Documentation/Nightly/Developers/Style_Guide
In this case:
 - Use 2 spaces for indents
 - Use brackets around your if/else
 - Do not add an extra return at the end of the function.

By the way, you should be able to call the openAddDataDialog() directly like this:
qSlicerApplication::application()->ioManager()->openAddDataDialog(); I think it's fine to expect the application to always have a valid ioManager.